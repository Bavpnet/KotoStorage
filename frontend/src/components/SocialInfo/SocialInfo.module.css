@value mobile-max-width from './../../globals.css';

.inSidebar {
  display: none;
}

.inFooter {
  display: flex;
  flex-direction: column;
  gap: 12px;
  justify-content: center;
  align-items: center;
}

.social {
  display: flex;
  list-style: none;
  gap: 26px;
  justify-content: center;
  align-items: center;
  padding: 0;
}

.infoTitle {
  font-size: 1.3rem;
  font-weight: 700;
}

.infoDesc {
  font-size: 0.6rem;
  max-width: 270px;
}

.social li {
  font-size: 0.58rem;
  font-family: 'Rubik', sans-serif;
  font-weight: 500;
}

@media only screen and (max-width: mobile-max-width) {
  .inSidebar {
    display: flex;
    flex-direction: column;
    width: 100%;
    gap: 12px;
    justify-content: center;
    align-items: center;
  }
}
