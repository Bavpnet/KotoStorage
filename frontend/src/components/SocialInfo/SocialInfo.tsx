import styles from './SocialInfo.module.css'

export const SocialInfo = () => {
  const socialButtons = ['Instagram', 'Telegram', 'Github']
  return (
    <div className={styles.info}>
      <p className={styles.info__title}>Follow us in the media</p>
      <p className={styles.info__desc}>
        Hear about exclusive news and changes in the cats world
      </p>
      <ul className={styles.social}>
        {socialButtons.map(soc => (
          <li key={`soc__${soc}`}>{soc}</li>
        ))}
      </ul>
    </div>
  )
}
