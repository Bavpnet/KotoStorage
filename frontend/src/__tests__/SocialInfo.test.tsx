import renderer from 'react-test-renderer'
import { SocialInfo } from '../components/SocialInfo'

describe('SocialInfo', () => {
  it('renders correctly footer', () => {
    const tree = renderer.create(<SocialInfo isFooterInfo={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly navbar', () => {
    const tree = renderer.create(<SocialInfo isFooterInfo={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
