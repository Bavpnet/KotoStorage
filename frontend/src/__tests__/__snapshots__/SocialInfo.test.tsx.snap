// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialInfo renders correctly footer 1`] = `
<section
  className="inFooter"
>
  <h2
    className="infoTitle"
  >
    Follow us in the media
  </h2>
  <p
    className="infoDesc"
  >
    Hear about exclusive news and changes in the cats world
  </p>
  <ul
    className="social"
  >
    <li>
      Instagram
    </li>
    <li>
      Telegram
    </li>
    <li>
      Github
    </li>
  </ul>
</section>
`;

exports[`SocialInfo renders correctly navbar 1`] = `
<section
  className="inSidebar"
>
  <h2
    className="infoTitle"
  >
    Follow us in the media
  </h2>
  <p
    className="infoDesc"
  >
    Hear about exclusive news and changes in the cats world
  </p>
  <ul
    className="social"
  >
    <li>
      Instagram
    </li>
    <li>
      Telegram
    </li>
    <li>
      Github
    </li>
  </ul>
</section>
`;
